packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -247,19 +247,21 @@ export const useGridFilter = (
     (params) => {
       if (props.filterMode === GridFeatureModeConstant.client && params.isRowMatchingFilters) {
         const rowIds = gridRowIdsSelector(apiRef.current.state);
-        const visibleRowsLookup: Record<GridRowId, boolean> = {};
+        const filteredRowsLookup: Record<GridRowId, boolean> = {};
         for (let i = 0; i < rowIds.length; i += 1) {
           const rowId = rowIds[i];
-          visibleRowsLookup[rowId] = params.isRowMatchingFilters(rowId);
+          filteredRowsLookup[rowId] = params.isRowMatchingFilters(rowId);
         }
         return {
-          visibleRowsLookup,
+          filteredRowsLookup,
+          visibleRowsLookup: filteredRowsLookup,
```

Comment:
So rows that are collapsed will be included in `visibleRowsLookup`, but with value set to `false`?

Reply:
The behavior of `visibleRowsLookupè remains the same
The collapsed rows are included with value set to `false` yes

Reply:
When reading it alone, it feels like "Why are you saving two times the same thing?"
A comment such as "visibleRowsLookup will be modified by the treeData and rowGrouping hook" could help to keep in mind which part of the code will reuse the difference between them.

By the way, don't you need to use `visibleRowsLookup: {...filteredRowsLookup}` to avoid side effects on `filteredRowsLookup` when modifying `visibleRowsLookup`?

Reply:
I'll add a comment to explain why we return the same thing twice :+1:

Reply:
If you do not double the memory consumption, how do you do to double the number of information per row?

The only solution to avoid useless consumption of memory I see is to do the copy in tree data function, such that if tree data and grouping is not used, the objects are the same (which is not a problem)

Reply:
For the flat tree, I dont do `visibleRowsLookup: {...filteredRowsLookup}`, that way I don't increase the memory used.
For non-flat tree, I do double the memory used, but as you say, the two lookups being different it's logical.

The line we are commenting is only used for the flat tree.